src/users/cdf-manage.vue

```diff
@@ -0,0 +1,236 @@
+      },
+    },
+    methods: {
+      getDojo(dojoId) {
+        return this.dojos.find(dojo => dojo.id === dojoId);
+      },
+      isDojoOwnerOf(membership) {
+        return !!membership.owner;
+      },
+      isChampionOf(membership) {
+        return membership.userTypes.indexOf('champion') > -1;
+      },
+      searchUser() {
+        return this.$router.push({ name: 'CDFUsersManagement', query: { email: this.email } });
+      },
+      async getUserInfos() {
+        this.errors.clear();
+        this.reset();
+        try {
+          if (this.email && this.email.length > 0) {
+            this.user = (await UserService.search({ email: this.email, related: 'profile' })).body;
+          } else if (this.userId && this.userId.length > 0) {
+            this.user = (await UserService.load(this.userId, { related: 'profile' })).body;
+          }
+          await Promise.all([
+            UserService.getChildren(this.user.id),
+            (async () => {
+              this.memberships = (await DojoService.getUsersDojos(this.user.id)).body;
+              return Promise.all(this.memberships.map(d =>
+                DojoService.getDojoById(d.dojoId)));
+            })(),
+          ]).then((res) => {
+            this.children = res[0].body;
+            this.dojos = res[1].map(resp => resp.body);
+          })
+          // Split this function from the promise.all : CORS errors are hard to catch
+          .then((async () => {
+            if (this.user.email) {
+              // TODO : once the forum integration is done back-end wise
+              // we can fully automate from there
+              try {
+                this.forumUser = (await ForumService.user.search(this.user.email)).body;
+              } catch (e) {
+                // You might not have the forums running in development; and that's fine
+                this.errors.add('forumUserNotFound', 'User not found');
+              }
+            }
+          })());
+        } catch (e) {
+          if (e.status === 404 && e.body.message === '404 - "Invalid userId"') {
+            this.errors.add('userNotFound', e.body.message);
+          }
+        }
+      },
+      async deleteUser(soft) {
+        this.errors.clear();
+        // eslint-disable-next-line no-alert
+        if (window.confirm(`Are you 100% sure you know what you are about to do to this poor ${this.user.name} ?`)) {
+          try {
+            await UserService.delete(this.user.id, !!soft);
+            this.deleted = true;
+          } catch (e) {
+            this.errors.add('deletionFailed', 'Something went absolutly wrong');
+          }
+        }
+      },
+      reset() {
+        this.user = {};
+        this.children = [];
+        this.dojos = [];
+        this.memberships = [];
+        this.deleted = false;
+      },
+    },
+    async created() {
+      this.reset();
+      if (!this.isLoggedIn) return this.$router.replace('/cdf');
```

Comment:
Shouldn't be needed with the nav guard, right?

Reply:
Very likely a relict of before I implemented the CDF navguard, yes
Good catch :+1: